docs/proposals/002-inference-gateway-support/proposal.md

```diff
@@ -0,0 +1,65 @@
+# Inference Gateway API Extension Support
+
+## Propsal Summary
+
+This proposal extends Envoy AI Gateway (EAIG) to support the Gateway API Inference Extension (GAIE), enabling seamless integration of both egress (external AI providers) and ingress (internal Kubernetes LLM deployments) traffic management.
+
+By allowing `AIServiceBackend.BackendRef` to reference GAIE's `InferencePool`, we unlock advanced capabilities like intelligent load balancing with fallbacks while maintaining a unified API.
+
+This strategic enhancement addresses user's needs without disrupting existing implementations, positioning EAIG as a comprehensive solution for all AI traffic management.
+
+**This document is about:**
+
+- Expanding the EAIG project scope to support GAIE
+- How it integrates with the existing EAIG APIs
+- Proposed implementation details
+
+## Background
+
+[Gateway API Inference Extension](https://gateway-api-inference-extension.sigs.k8s.io) (GAIE) is a new extension of the Kubernetes Gateway API that addresses serving large language models (LLMs) inside Kubernetes, with particular focus on intelligent load balancing decisions.
+
+Envoy AI Gateway (EAIG) was initially designed to route traffic to different AI providers, serving primarily as an egress solution.
+
+The EAIG MVP features targeted application developers with an egress focus, while the GAIE serves AI platform teams with a Kubernetes ingress focus.
```

Comment:
For someone who is deploying models solely on kubernetes, is there a benefit to use EAIG?

Reply:
I think even right now without GAIE implementation, the rate limit and the metrics/monitoring feature are useful to those without egress plus we are looking to expand the request transformation pluggable so that local hosted models can host whatever endpoint they want. These existing API/features will be available to GAIE used via AIServiceBackend.BackendRef. cc @yuzisun @missBerg for additional comments